kotlin_spark/src/main/java/com/sleazyweasel/eboshi/Auth.kt

```diff
@@ -0,0 +1,68 @@
+package com.sleazyweasel.eboshi
+
+import org.jose4j.jwk.RsaJsonWebKey
+import org.jose4j.jwk.RsaJwkGenerator
+import org.jose4j.jws.AlgorithmIdentifiers
+import org.jose4j.jws.JsonWebSignature
+import org.jose4j.jwt.JwtClaims
+import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import org.mindrot.jbcrypt.BCrypt
+
+private val TOKEN_ISSUER = "Kotlin Eboshi API"
+private val TOKEN_AUDIENCE = "eboshi user"
+private val EMAIL_CLAIM_KEY = "email"
+
+open class Auth {
+    private val rsaWebKey: RsaJsonWebKey;
+
+    constructor() {
+        rsaWebKey = RsaJwkGenerator.generateJwk(2048)!!
```

Comment:
It is a singleton Auth, only because it's only used in one place in the app, and only injected once. Yes, in the real world, this would not want to be sitting in here, but externalized.  I could also mark it as a singleton in the guice module (EboshiModule.kt), if I wanted to enforce that at the guice level.

Basically, I was being somewhat lazy, since it's a toy implementation. But, it does have the advantage of not requiring anything fancy or creds being stored in the repository/environment.